plugins/filter/time.py

```diff
@@ -130,7 +133,33 @@ def to_years(human_time, **kwargs):
     return to_time_unit(human_time, 'y', **kwargs)
 
 
-class FilterModule(object):
+def time_delta(date_time, **kwargs):
+    ''' Return datetime after calculating time delta '''
+    if not isinstance(date_time, str):
+        raise AnsibleFilterError('time_delta accepts datetime in string format')
+    date_format = "%Y-%m-%d %H:%M:%S"
+    if 'date_format' in kwargs:
+        date_format = kwargs.get('date_format')
+
+    delta = {
+        'days': kwargs.get('days', 0),
+        'microseconds': kwargs.get('microseconds', 0),
+        'milliseconds': kwargs.get('milliseconds', 0),
+        'minutes': kwargs.get('minutes', 0),
+        'hours': kwargs.get('hours', 0),
+        'weeks': kwargs.get('weeks', 0),
+    }
```

Comment:
It would be good to validate the above to be integers if supplied as options. Also you should check whether `kwargs` has more arguments than the ones we use (and fail if there are more).